src/ffprobe.rs

```diff
@@ -85,11 +78,24 @@ pub fn probe(input: &Path) -> Ffprobe {
         has_audio,
         max_audio_channels,
         resolution,
-        has_image_extension,
+        is_image,
         pix_fmt,
     }
 }
 
+fn is_image(path: &Path) -> anyhow::Result<bool> {
+    // This code was taken from infer::get_from_path, but it's not available since we're importing it as no_std
+    let file = File::open(path)?;
+    let limit = file
+        .metadata()
+        .map(|m| std::cmp::min(m.len(), 8192) as usize + 1)
+        .unwrap_or(0);
+    let mut file_header = Vec::with_capacity(limit);
+    file.take(8192).read_to_end(&mut file_header)?;
```

Comment:
I know you've taken this code from _infer_ but it is a bit odd.

I imagine querying metadata will be slower than any micro gain from allocating a smaller number than 8K, which we don't really expect anyway.

```suggestion
    let file = File::open(path)?;
    let mut file_header = Vec::with_capacity(8192);
    file.take(8192).read_to_end(&mut file_header)?;
```